hack/bundle-gen.py

```diff
@@ -0,0 +1,429 @@
+    print("Copying bundle directory")
+    bundle_files = os.path.join(bundle_source_dir, new_version)
+    hive_dir = os.path.join(repo_full_path, HIVE_SUB_DIR, new_version)
+    shutil.copytree(bundle_files, hive_dir)
+
+    # update bundle manifest
+    print("Updating bundle manfiest")
+    bundle_manifests_file = os.path.join(repo_full_path, HIVE_SUB_DIR, "hive.package.yaml")
+    bundle = {}
+    with open(bundle_manifests_file, 'r') as a_file:
+        bundle = yaml.load(a_file, Loader=yaml.SafeLoader)
+
+    found = False
+    for channel in bundle["channels"]:
+        if channel["name"] in update_channels:
+            found = True
+            channel["currentCSV"] = "hive-operator.{}".format(new_version)
+
+    if not found:
+        print("did not find a CSV channel to update")
+        sys.exit(1)
+
+    with open(bundle_manifests_file, 'w') as outfile:
+        yaml.dump(bundle, outfile, default_flow_style=False)
+    print("\nUpdated bundle package:\n\n")
+    cmd = ("cat %s" % bundle_manifests_file).split()
+    subprocess.run(cmd)
+    print()
+
+    # commit files
+    print("Adding file")
+    repo.git.add(HIVE_SUB_DIR)
+
+    print("Commiting {}".format(pr_title))
+    try:
+        repo.git.commit('--signoff', '--message="{}"'.format(pr_title))
+    except:
+        print("Failed to commit")
+        raise
+    print()
+
+    if not dry_run:
+        print("Pushing branch {}".format(branch_name))
+        origin = repo.git.origin
+        try:
+            origin.push('--force', branch_name)
+        except:
+            print("failed to push branch to origin")
+            raise
+
+        # open PR
+        gh = importlib.import_module('github')
+        client = gh.GitHubClient(dest_github_org, dest_github_reponame, "")
+
+        from_branch = "{}:{}".format(gh_username, branch_name)
+        to_branch = "main"
+
+        body = pr_title
+        if hold:
+            body = "%s\n\n/hold" % body
+
+        resp = client.create_pr(from_branch, to_branch, pr_title, body)
+        if resp.status_code != 201: #201 == Created
+            print(resp.text)
+            sys.exit(1)
+
+        json_content = json.loads(resp.content.decode('utf-8'))
+        print("PR opened: {}".format(json_content["html_url"]))
+
+    else:
+        print("Skipping branch push due to dry-run")
+    print()
+
+def branch_is_on_master_commit(branch, repo_dir):
+    repo = git.Repo(repo_dir)
+    return repo.rev_parse('master') == repo.rev_parse(branch)
+
+def buildah_installed():
+    return shutil.which('buildah') is not None
+
+if __name__ == "__main__":
+    args = get_params()
+
+    if not buildah_installed():
+        print('buildah not found, please install buildah')
+        sys.exit(1)
+
+    update_channels = [CHANNEL_DEFAULT]
+    if args.branch:
+        if branch_is_on_master_commit(args.branch, os.getcwd()):
+            update_channels.append(args.branch)
+        else:
+            update_channels = [args.branch]
```

Comment:
This is nice. But it won't work if I specify a commit hash. So we should probably check for that. (Not sure how.)